apps/speed.c

```diff
@@ -197,6 +212,42 @@ static volatile int run = 0;
 static int mr = 0;
 static int usertime = 1;
 
+typedef struct loopargs_st {
+    ASYNC_JOB *inprogress_job;
+    EVP_CIPHER_CTX ctx;
+} loopargs;
+
+int EVP_Digest_MD2_loop(void *args);
+int EVP_Digest_MDC2_loop(void *args);
+int EVP_Digest_MD4_loop(void *args);
+int MD5_loop(void *args);
+int HMAC_loop(void *args);
+int SHA1_loop(void *args);
+int SHA256_loop(void *args);
+int SHA512_loop(void *args);
+int WHIRLPOOL_loop(void *args);
+int EVP_Digest_RMD160_loop(void *args);
+int RC4_loop(void *args);
+int DES_ncbc_encrypt_loop(void *args);
+int DES_ede3_cbc_encrypt_loop(void *args);
+int AES_cbc_128_encrypt_loop(void *args);
+int AES_cbc_192_encrypt_loop(void *args);
+int AES_ige_128_encrypt_loop(void *args);
+int AES_cbc_256_encrypt_loop(void *args);
+int AES_ige_192_encrypt_loop(void *args);
+int AES_ige_256_encrypt_loop(void *args);
+int CRYPTO_gcm128_aad_loop(void *args);
+int EVP_Update_loop(void *args);
+int EVP_Digest_loop(void *args);
+int RSA_sign_loop(void *args);
+int RSA_verify_loop(void *args);
+int DSA_sign_loop(void *args);
+int DSA_verify_loop(void *args);
+int ECDSA_sign_loop(void *args);
+int ECDSA_verify_loop(void *args);
+int ECDH_compute_key_loop(void *args);
+int run_benchmark(int async_jobs, loopargs **array_loopargs, int (*loop_function)(void *));
```

Comment:
Yes, they can all definitely be all `static` so I will change them.

Also, I think I should add the `#ifndef` around each declaration.
For example:
```c
#ifndef OPENSSL_NO_MD2
static int EVP_Digest_MD2_loop(void *args);
#endif
```

Would you prefer to have these moved just before the definitions instead of the beginning of the file?